Code/Source/URDF/FbxNode.cpp

```diff
@@ -0,0 +1,161 @@
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "FbxNode.h"
+
+#include <fstream>
+#include <string>
+
+#include <AzCore/Debug/Trace.h>
+
+namespace ROS2
+{
+    namespace Fbx
+    {
+        std::string AnyToString(const std::any & a)
+        {
+            if (a.type() == typeid(int))
+            {
+                return std::to_string(std::any_cast<int>(a));
+            }
+            else if (a.type() == typeid(unsigned))
+            {
+                return std::to_string(std::any_cast<unsigned>(a));
+            }
+            else if (a.type() == typeid(float))
+            {
+                return std::to_string(std::any_cast<float>(a));
+            }
+            else if (a.type() == typeid(double))
+            {
+                return std::to_string(std::any_cast<double>(a));
+            }
+            else if (a.type() == typeid(const char*))
+            {
+                return std::string("\"") + std::any_cast<const char*>(a) + std::string("\"");
+            }
+            else if (a.type() == typeid(std::string))
+            {
+                return std::string("\"") + std::any_cast<std::string>(a) + std::string("\"");
+            }
+            else if (a.type() == typeid(RawString))
+            {
+                return std::any_cast<RawString>(a).data;
+            }
+
+            return "";
+        }
+
+        Node::Node(const std::string & name, const Properties & properties)
+            : m_name(name)
+            , m_properties(properties)
+        {}
+
+        std::string Node::GetName() const
+        {
+            return m_name;
+        }
+
+        std::vector<Node> Node::GetChildren() const
+        {
+            return m_children;
+        }
+
+        Properties Node::GetProperties() const
+        {
+            return m_properties;
+        }
+
+        bool Node::HasChildren() const
+        {
+            return !m_children.empty();
+        }
+
+        bool Node::HasProperties() const
+        {
+            return !m_properties.empty();
+        }
+
+        void Node::AddProperty(const Property & property)
+        {
+            m_properties.push_back(property);
+        }
+
+        void Node::AddChild(const Node & child)
+        {
+            m_children.push_back(child);
+        }
+
+        void Node::AddChild(const std::string & name, const Property & property)
+        {
+            m_children.push_back(Node(name, { property }));
+        }
+
+        void Node::AddChild(const Node && child)
+        {
+            m_children.push_back(child);
+        }
+
+        std::string Node::ToString(int nodeDepth) const
+        {
+            std::stringstream ss;
+            std::string offset;
+
+            // Calculate offset
+            for (int i = 0; i < nodeDepth; ++i)
+                offset += "  ";
```

Comment:
FYI:
You can use the std::setfill and std::setw into a another string stream
(You will need to add ```#include <iomanip>``` at the top of this file)
```suggestion
            // Calculate offset
            std::stringstream offset_stream;
            offset_stream << std::setfill('  ') << std::setw(nodeDepth) << "";
            std::string offset = offset_stream.str();
```